javascript-bank/bank.js

```diff
@@ -0,0 +1,40 @@
+/* exported Bank */
+
+function Bank() {
+  this.nextAccountNumber = 1;
+  this.accounts = [];
+}
+
+Bank.prototype.openAccount = function (holder, balance) {
+  if (balance <= 0 || Number.isInteger(balance) === false) {
+    return null;
+  }
+  var newAccount = new Account(this.nextAccountNumber, holder);
+  newAccount.deposit(balance);
+  this.accounts.push(newAccount);
+  this.nextAccountNumber++;
+  return newAccount.number;
+};
+
+Bank.prototype.getAccount = function (number) {
+  if (number > this.accounts.length) {
+    return null;
+  }
+  for (var i = 0; this.accounts.length; i++) {
+    if (this.accounts[i].number === number) {
+      return this.accounts[i];
+    }
+  }
+  return null;
+};
+
+Bank.prototype.getTotalAssets = function () {
+  var total = 0;
+  if (this.accounts.length === 0) {
+    return 0;
+  }
```

Comment:
This conditional shouldn't be necessary. If the length is `0`, your loop would never start, so you would still end up returning `0` after the loop attempts to run